tony-core/src/main/java/com/linkedin/tony/TonyJobMetadata.java

```diff
@@ -38,21 +35,26 @@ private TonyJobMetadata(String id, String url, long started, long completed, Str
     this.user = user;
   }
 
-  public static TonyJobMetadata newInstance(Configuration yarnConf, String appId, long started, long completed,
-      boolean status, String user) {
-    String jobStatus = status ? "SUCCEEDED" : "FAILED";
+  // Since newInstance could be called when the job hasn't finished,
+  // `completed` and `status` could be null
+  public static TonyJobMetadata newInstance(Configuration yarnConf, String appId, Long started, Long completed,
```

Comment:
Another pattern to consider is using a Builder to construct a TonyJobMetadata. Example:

```
// After defining a static class Builder class inside TonyJobMetadata
new TonyJobMetadata.Builder().setConf(new Configuration()).setAppId(...).setStartTime(...).setStatus().setUser().build();
```

If something isn't set (e.g.: endTime), then it can default to some value like `-1L`. One advantage of this approach is it's less error-prone to accidentally swapping the arguments (e.g.: swapping start and end time or appId and user). You also avoid code like `TonyJobMetadata.newInstance(yarnConf, appIdString, started, null, null, user)`, from which you can't tell from a glance what the `null`s mean.